containers/src/Kokkos_DynRankView.hpp

```diff
@@ -580,9 +580,37 @@ class DynRankView : private View<DataType*******, Properties...> {
              index_type i6 = 0) const {
     return view_type::operator()(i0, i1, i2, i3, i4, i5, i6);
   }
+
+// This is an accomodation for Phalanx, who use the operator[] to access all
+// memory
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   KOKKOS_FUNCTION reference_type operator[](index_type i0) const {
+    if constexpr (std::is_same<typename drvtraits::value_type,
+                               typename drvtraits::scalar_array_type>::value) {
```

Comment:
```suggestion
    if constexpr (std::is_same_v<typename drvtraits::value_type,
                               typename drvtraits::scalar_array_type>) {
```
(not formatted)

I thought we were enforcing this?!